tests/models/animals/test_animal_model.py

```diff
@@ -214,18 +214,24 @@ def test_calculate_litter_additions(functional_group_list_instance):
     model = AnimalModel(data, pint.Quantity("1 week"), functional_group_list_instance)
 
     # Update the waste pools
-    excess_excrement = [3.5e3, 5.6e4, 5.9e4, 2.3e6]
+    decomposed_excrement = [3.5e3, 5.6e4, 5.9e4, 2.3e6]
     for ind, community in enumerate(model.communities.values()):
-        community.excrement_pool.stored_energy = excess_excrement[ind]
+        community.excrement_pool.decomposed_energy = decomposed_excrement[ind]
```

Comment:
I *think* this could be:

```python
    for energy, community in zip(decomposed_excrement, model.communities.values()):
        community.excrement_pool.decomposed_energy = energy
```

Reply:
Yep that works! Is it generally better to make use of `zip()` rather than enumerate + indexing?

Reply:
In general it's best practice to avoid looping over indices if there's another way, because a) it's usually more readable and b) it stops you from making off-by-one errors etc.

I've realised that these two things aren't functionally identical though. `zip` is fine if you know your sequences have the same length, otherwise it'll just stop after either one of them is finished:

```python
>>> list(zip(range(2),range(3)))
[(0, 0), (1, 1)]
```

I guess that's fine here though?

Reply:
Yeah I think that's fine here, like if the number of communities returned changes that is something that should have been caught by a different unit test